repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala

```diff
@@ -55,6 +57,14 @@ class ExecutorClassLoader(conf: SparkConf, classUri: String, parent: ClassLoader
     }
   }
 
+  override def getResource(name: String): URL = {
+    parentLoader.getResource(name)
```

Comment:
I'd be surprised if the repl generated resources, but at the same time, what if someone tries to load the generated class file as a resource? It's an unusual but valid use case.

The code to support that is not complicated; it already exists in `MutableURLClassLoader.scala` (class `ChildFirstURLClassLoader` for, example). If you're up for it you could even create a helper trait or a utility method somewhere for that.